api/server/server.go

```diff
@@ -35,28 +35,37 @@ const (
 func initDependencies(config Config) error {
 	// ensure all initialization code fails fast on errors; there is no point in
 	// attempting to continue in a degraded state if there are problems at start up
-	if err := initEtcd(config); err != nil {
-		return err
-	}
-	if err := initElasticsearch(config); err != nil {
-		return err
-	}
-	if err := initNats(config); err != nil {
-		return err
-	}
-	if err := initInfluxDB(config); err != nil {
-		return err
-	}
-	if err := initDocker(config); err != nil {
-		return err
+
+	errors := make(chan error)
+	go func() {
+		errors <- initEtcd(config)
+	}()
+	go func() {
+		errors <- initElasticsearch(config)
+	}()
+	go func() {
+		errors <- initNats(config)
+	}()
+	go func() {
+		errors <- initInfluxDB(config)
+	}()
+	go func() {
+		errors <- initDocker(config)
+	}()
+
+	for i := 0; i < 5; i++ { // 5 is not a magic number, it corresponds to the number of concurrent go routines
```

Comment:
@generalhenry why not, it's a different but maybe more orthogonal approach. I'll update my code to use it.